specs/sharding/beacon-chain.md

```diff
@@ -627,12 +707,19 @@ def process_shard_header(state: BeaconState, signed_header: SignedShardBlobHeade
         == bls.Pairing(body_summary.commitment.point, G2_SETUP[-body_summary.commitment.length])
     )
 
+    # Charge builder, with hard balance requirement
+    fee = Gwei(123)  # TODO EIP 1559 like fee? Burn some of it?
```

Comment:
Builder will need to be able to set a fee inside of the `ShardBlobHeader`. It must at least have the 1559 fee for burn and any excess goes to the proposer

Reply:
and if blob isn't confirmed, just a fraction should go to proposer